Library/Homebrew/utils/shell.rb

```diff
@@ -0,0 +1,64 @@
+require 'shellwords'
+
+module Utils
+  module Shell
+    # take a path and heuristically convert it
+    # to a shell, return nil if there's no match
+    def self.path_to_shell(path)
+      # remove leading - from login shell
+      shell_name = File.basename(path).strip_prefix("-")
+      case shell_name
+      when "sh", "bash", "zsh", "csh", "tcsh", "fish", "ksh"
+        shell_name.to_sym()
+      else nil
+      end
```

Comment:
The `else nil` case seems redundant. You could also replace the entire `case` statement with the following line:

```ruby
shell_name.to_sym if %w[bash csh fish ksh sh tcsh zsh].include?(shell_name)
```